cache/global_lock.go

```diff
@@ -81,6 +82,23 @@ func startLock() error {
 		}
 	}()
 
+	// Capture ctrl-c or other interruptions then clean up the global lock.
+	ch := make(chan os.Signal)
+	signal.Notify(ch, os.Interrupt, os.Kill)
+	go func(cc <-chan os.Signal) {
+		s := <-cc
+		SystemUnlock()
```

Comment:
Seems racy to me. Nothing else gets blocked on the signal, so after you unlock it, another go routine could potentially acquire (create) a lock after this call, and before the exit.